pkg/instance/handler.go

```diff
@@ -167,6 +181,39 @@ func (h Handler) Deploy(c *gin.Context) {
 	c.JSON(http.StatusAccepted, savedInstance)
 }
 
+func (h Handler) consumeParameters(user *models.User, sourceInstanceId uint, instance *model.Instance, preset bool) error {
+	sourceInstance, err := h.instanceService.FindByIdDecrypted(sourceInstanceId)
+	if err != nil {
+		return err
+	}
+
+	if preset && !sourceInstance.Preset {
+		return apperror.NewUnauthorized(fmt.Sprintf("instance (id: %d) isn't a preset", sourceInstance.ID))
+	}
+
+	if preset && sourceInstance.StackName != instance.StackName {
+		return apperror.NewUnauthorized(fmt.Sprintf("preset stack (%s) doesn't match instance stack (%s)", sourceInstance.StackName, instance.StackName))
```

Comment:
Maybe we should default to `%q` for printing strings. We actually use it regularly. Maybe not always. That saves you from doing your own quoting or highlighting via `()`.

Reply:
Good catch!